### ISSUE TITLE
Request Headers should convert object to Headers

### ISSUE THREAD

**Issue by gauntface** [pinterest/service-workers#65]
The request will currently accept headers as an object and apply them directly when they should be converted to a Headers instance.

This makes headers an object:
```javascript
const request = new Request(requestUrl, {
    header, {
        'x-foo': 'bar',
    }
});
```

This makes headers a Headers Map:
```
const headers = new Headers();
headers.set('x-foo', 'bar');

const request = new Request(requestUrl, {
    headers,
});
```

**Comment by philipwalton** [comment]
(Related to #48)

In addition to what @gauntface mentioned:

- `Headers` has an `append()` method that's not the same as `set()`
- `Headers` can be constructed from a `Headers` object, which is different from map
- `Map` has a few methods that `Headers` doesn't have, like `clear()`.

I'd recommend implementing `Headers` as composing `Map` rather than extending `Map`.

**Comment by zackargyle** [comment]
Oh no! My laziness is showing! I'll get this fixed for you all 😳

**Comment by philipwalton** [comment]
Thanks! BTW, we have some mocks that we're using locally that we'll try to upstream once we're happy with them.

**Comment by zackargyle** [comment]
@philipwalton mind taking a look at the PR above ^? Should work, and my tests pass, but wanted to make sure before I push and publish again!

**Comment by zackargyle** [comment]
The fix has been published to `1.7.1`. Thanks for report!
